src/seedu/addressbook/data/person/Address.java

```diff
@@ -6,53 +6,30 @@
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
-public class Address {
+public class Address extends Contact {
 
     public static final String EXAMPLE = "123, some street";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
-    public final String value;
-    private boolean isPrivate;
-
     /**
      * Validates given address.
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
+        if (!isValid(address, ADDRESS_VALIDATION_REGEX)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
```

Comment:
As discussed in class, you can add a constructor in your inherited class and do `super()`